setup.sh

```diff
@@ -43,52 +43,54 @@ declare -r CONFIGS_PATH='configs'
 
 declare -r DOCS_VIRTUAL_ENV='docs_virtual_env'
 
+function debian_pkg_check()
+{
+  dpkg-query -W --showformat='${Status}\n' "$1" 2> /dev/null | grep -c 'ok installed'
+}
+
 function check_dependencies()
 {
   local package_list=''
   local cmd=''
   local distro
-
+  readonly IS_INSTALLED_CMD=0
+  readonly DISTRO_CMD_FOR_INSTALL_PACKAGE=1
+  declare -a debian=('debian_pkg_check ' 'apt install -y')
+  declare -a arch=('pacman -Ql ' 'pacman -S')
+  declare -a gentoo=('qlist -I ' 'emerge')
+  declare -a fedora=('rpm -q ' 'dnf install -y')
+  declare -A distro_commands=(['arch']=arch ['gentoo']=gentoo ['fedora']=fedora ['debian']=debian)
+  declare -n distro_cmd
   distro=$(detect_distro '/')
+  distro_cmd="${distro_commands["$distro"]}"
 
-  if [[ "$distro" =~ 'arch' ]]; then
-    while IFS='' read -r package; do
-      installed=$(pacman -Ql "$package" &> /dev/null)
-      [[ "$?" != 0 ]] && package_list="$package $package_list"
-    done < "$DOCUMENTATION/dependencies/arch.dependencies"
-    cmd="pacman -S $package_list"
-  elif [[ "$distro" =~ 'debian' ]]; then
+  if [[ -v distro_commands["$distro"] ]]; then
```

Comment:
Can't we make `if [[ -n "${distro_commands["$distro"]}" ]]`. I get that you are checking set/unset, but I think, in this case, that checking empty (using `-n`) and unset are virtually the same (with the former being more intuitive and allowing a safer expansion).

Reply:
yes I'll change that